src/transcode.jl

```diff
@@ -76,9 +80,11 @@ julia> String(decompressed)
 
 ```
 """
-function Base.transcode(codec::Codec, data::ByteData)
-    input = Buffer(data)
-    output = Buffer(initial_output_size(codec, buffermem(input)))
+function Base.transcode(
+    codec::Codec,
+    input::Buffer,
+    output::Buffer = Buffer(initial_output_size(codec, buffermem(input))),
+)
```

Comment:
This is mainly cosmetic, but it is easier to read this way

```suggestion
_default_output_buffer(codec, input) = Buffer(
    initial_output_size(
        codec,
        buffermem(input)
    )
)

function Base.transcode(
    codec::Codec,
    input::Buffer,
    output::Buffer = _default_output_buffer(codec, input) ,
)
```